datafusion/execution/src/memory_pool/pool.rs

```diff
@@ -163,25 +242,30 @@ impl MemoryPool for FairSpillPool {
     }
 
     fn unregister(&self, consumer: &MemoryConsumer) {
+        let mut state = self.state.lock();
+        let used = state.meta.unregister(consumer.name());
+
         if consumer.can_spill {
-            let mut state = self.state.lock();
-            state.num_spill = state.num_spill.checked_sub(1).unwrap();
+            state.num_spill = state.num_spill.saturating_sub(1);
```

Comment:
I wonder if there is a reason to use `saturating_sub` rather than `checked_sub`?